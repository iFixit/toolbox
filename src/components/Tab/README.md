### Examples

#### Default Tab
```
<Tab>All</Tab>
```

#### Active Tab
```
<Tab active>All</Tab>
```

#### Element prop

The `Tab` component accepts an `is` prop to change the underlying HTML element on a per-instance basis. This is useful for ensuring semantic markup, while keeping styles decoupled.

```
<Tab is="span">All</Tab>
```