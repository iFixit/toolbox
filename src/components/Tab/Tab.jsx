import React from 'react';
import { bool, element, func, string, oneOfType } from 'prop-types';
import styled from 'styled-components';

import constants from '../../constants';

const { spacing, fontSize, lineHeight, color, transition } = constants;

const TabContainer = styled.a(
   {
      flex: '0 0 auto',
      display: 'inline-block',
      boxSizing: 'border-box',
      padding: spacing[2],
      marginRight: spacing[2],
      fontSize: fontSize[2],
      fontWeight: 700,
      lineHeight: lineHeight.title,
      textDecoration: 'none',
      color: color.grayAlpha[9],
      borderTop: '3px solid transparent',
      borderBottom: '3px solid transparent',
      transition: `all ${transition.duration} ${transition.easing}`,
      outline: 'none',

      '&:hover': {
         color: color.blue[4],
      },

      '&:focus': {
         borderBottomColor: color.grayAlpha[4],
      },
   },
   ({ active }) =>
      active && {
         color: color.blue[4],
         borderBottomColor: color.blue[4],
      },
);

const Tab = ({ is, ...props }) => {
   const Component = TabContainer.withComponent(is);
   return <Component {...props} />;
};

Tab.propTypes = {
   active: bool,
   /** HTML element string or React component to render */
   is: oneOfType([string, func, element]),
};

Tab.defaultProps = {
   active: false,
   is: 'a',
};

export default Tab;
