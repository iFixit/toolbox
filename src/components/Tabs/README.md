### Example

```
<Tabs>
  <Tab active>All</Tab>
  <Tab>Store</Tab>
  <Tab>Guides</Tab>
</Tabs>
```