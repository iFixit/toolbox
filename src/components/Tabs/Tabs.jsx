import React from 'react';
import { arrayOf, node } from 'prop-types';
import glamorous from 'glamorous';

import constants from '../../constants';

const { color, spacing } = constants;

const Container = glamorous.div({
   position: 'relative',
   borderBottom: `1px solid ${color.grayAlpha[3]}`,

   ':after': {
      content: '""',
      display: 'block',
      position: 'absolute',
      top: 0,
      bottom: 0,
      right: 0,
      width: spacing[2],
      background:
         `linear-gradient(to left, ${color.white}, rgba(255, 255, 255, 0))`,
   },
});

const ScrollContainer = glamorous.div({
   display: 'flex',
   alignItems: 'center',
   overflowX: 'auto',
   overflowY: 'hidden',
   WebkitOverflowScrolling: 'touch',
});

const TabsWrapper = glamorous.div({
   display: 'flex',
});

const Tabs = ({ children, ...rest }) => (
   <Container {...rest}>
      <ScrollContainer>
         <TabsWrapper>
            {children}
         </TabsWrapper>
      </ScrollContainer>
   </Container>
);

Tabs.propTypes = {
   children: arrayOf(node).isRequired,
};

export default Tabs;
