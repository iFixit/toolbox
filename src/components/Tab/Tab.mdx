---
name: Tab
route: /components/tab
menu: Components
---

import Tab from './Tab'
import { Playground, PropsTable } from 'docz'

# Tab

## Examples

### Default Tab
<Playground>
  <Tab>All</Tab>
</Playground>

### Active Tab
<Playground>
  <Tab active>All</Tab>
</Playground>

### Element prop

The `Tab` component accepts an `is` prop to change the underlying HTML element on a per-instance basis. This is useful for ensuring semantic markup, while keeping styles decoupled.

<Playground>
  <Tab is="span">All</Tab>
</Playground>

## Props

<PropsTable of={Tab} />
