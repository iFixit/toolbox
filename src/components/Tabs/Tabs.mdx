---
name: Tabs
route: /components/tabs
menu: Components
---

import Tabs from './Tabs'
import Tab from '../Tab/Tab'
import { Playground, PropsTable } from 'docz'

# Tabs

## Example

<Playground>
  <Tabs>
    <Tab active>All</Tab>
    <Tab>Store</Tab>
    <Tab>Guides</Tab>
    <Tab>Wikis</Tab>
    <Tab>Devices</Tab>
    <Tab>Answers</Tab>
  </Tabs>
</Playground>

## Props

<PropsTable of={Tabs} />
