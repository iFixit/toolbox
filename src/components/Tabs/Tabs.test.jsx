import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from './Tabs';
import Tab from '../Tab/Tab';

test('renders without crashing', () => {
   const tree = renderer
      .create(
         <Tabs>
            <Tab active>All</Tab>
            <Tab>Store</Tab>
            <Tab>Guides</Tab>
            <Tab>Wikis</Tab>
            <Tab>Devices</Tab>
            <Tab>Answers</Tab>
         </Tabs>,
      )
      .toJSON();
   expect(tree).toMatchSnapshot();
});
