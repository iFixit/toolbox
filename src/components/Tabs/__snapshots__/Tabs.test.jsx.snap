// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c0 {
  border-bottom: 1px solid rgba(0,3,6,0.12);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
  overflow-y: hidden;
  -webkit-overflow-scrolling: touch;
  -webkit-mask-image: linear-gradient(to left,transparent,#000000 12px);
  mask-image: linear-gradient(to left,transparent,#000000 12px);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  box-sizing: border-box;
  padding: 12px;
  margin-right: 12px;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0,3,6,0.87);
  border-top: 3px solid transparent;
  border-bottom: 3px solid transparent;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  outline: none;
  color: #0071CE;
  border-bottom-color: #0071CE;
}

.c3:hover {
  color: #0071CE;
}

.c3:focus {
  border-bottom-color: rgba(0,3,6,0.26);
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  box-sizing: border-box;
  padding: 12px;
  margin-right: 12px;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0,3,6,0.87);
  border-top: 3px solid transparent;
  border-bottom: 3px solid transparent;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  outline: none;
}

.c4:hover {
  color: #0071CE;
}

.c4:focus {
  border-bottom-color: rgba(0,3,6,0.26);
}

.c5 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  box-sizing: border-box;
  padding: 12px;
  margin-right: 12px;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0,3,6,0.87);
  border-top: 3px solid transparent;
  border-bottom: 3px solid transparent;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  outline: none;
}

.c5:hover {
  color: #0071CE;
}

.c5:focus {
  border-bottom-color: rgba(0,3,6,0.26);
}

.c6 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  box-sizing: border-box;
  padding: 12px;
  margin-right: 12px;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0,3,6,0.87);
  border-top: 3px solid transparent;
  border-bottom: 3px solid transparent;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  outline: none;
}

.c6:hover {
  color: #0071CE;
}

.c6:focus {
  border-bottom-color: rgba(0,3,6,0.26);
}

.c7 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  box-sizing: border-box;
  padding: 12px;
  margin-right: 12px;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0,3,6,0.87);
  border-top: 3px solid transparent;
  border-bottom: 3px solid transparent;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  outline: none;
}

.c7:hover {
  color: #0071CE;
}

.c7:focus {
  border-bottom-color: rgba(0,3,6,0.26);
}

.c8 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  box-sizing: border-box;
  padding: 12px;
  margin-right: 12px;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0,3,6,0.87);
  border-top: 3px solid transparent;
  border-bottom: 3px solid transparent;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  outline: none;
}

.c8:hover {
  color: #0071CE;
}

.c8:focus {
  border-bottom-color: rgba(0,3,6,0.26);
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <a
        className="c3"
      >
        All
      </a>
      <a
        className="c4"
      >
        Store
      </a>
      <a
        className="c5"
      >
        Guides
      </a>
      <a
        className="c6"
      >
        Wikis
      </a>
      <a
        className="c7"
      >
        Devices
      </a>
      <a
        className="c8"
      >
        Answers
      </a>
    </div>
  </div>
</div>
`;
